src/main/java/com/todolist/app/servlet/SeeAllTasksServlet.java

```diff
@@ -0,0 +1,48 @@
+package com.todolist.app.servlet;
+
+import com.todolist.app.model.Item;
+import com.todolist.app.model.Status;
+import com.todolist.app.service.ToDoStoreSingleton;
+import jakarta.servlet.*;
+import jakarta.servlet.http.*;
+import jakarta.servlet.annotation.*;
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+
+@WebServlet(name = "SeeAllTasksServlet", value = "/see-all-tasks")
+public class SeeAllTasksServlet extends HttpServlet {
+    static ToDoStoreSingleton singleton = ToDoStoreSingleton.getInstance();
+
+    static {
+        singleton.getExample();
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        Map<String, Item> onhold = new LinkedHashMap<>();
+        Map<String, Item> completed = new LinkedHashMap<>();
+        for(Map.Entry<String, Item> entry : singleton.getItems().entrySet()){
```

Comment:
Can we use the full power of Java 8 Stream API, e.g. `groupingBy`? See https://www.baeldung.com/java-groupingby-collector

Reply:
@aholtvianskyi As I understand with groupingBy I'll lose key values from map for tables, but they will be necessary in future in jsp file

Reply:
Oh, I see.

In any case the same can be done by using `partitionby`, https://www.javacodegeeks.com/2015/11/java-8-streams-api-grouping-partitioning-stream.html e.g.

```
Map<String, Item> items = new LinkedHashMap<>();
        items.put("1", new Item("1", "11-12-2023", Status.COMPLETED, Priority.NORMAL, Arrays.asList(Tag.READING)));
        items.put("2", new Item("2", "11-12-2023", Status.COMPLETED, Priority.NORMAL, Arrays.asList(Tag.READING)));
        items.put("3", new Item("3", "11-12-2023", Status.COMPLETED, Priority.NORMAL, Arrays.asList(Tag.READING)));
        items.put("4", new Item("4", "11-12-2023", Status.PENDING, Priority.NORMAL, Arrays.asList(Tag.READING)));
        items.put("5", new Item("5", "11-12-2023", Status.INPROGRESS, Priority.NORMAL, Arrays.asList(Tag.READING)));

Map<Boolean, List<Map.Entry<String, Item>>> collect = items.entrySet().stream().collect(
      partitioningBy(val -> val.getValue().getStatus() == Status.COMPLETED)
);

// [1=1, 11-12-2023, COMPLETED, NORMAL, [READING], 2=2, 11-12-2023, COMPLETED, NORMAL, [READING], 3=3, 11-12-2023, COMPLETED, NORMAL, [READING]]
System.out.println(collect.get(Boolean.TRUE)); // will print 3 elements which has corresponding completed status.

[4=4, 11-12-2023, PENDING, NORMAL, [READING], 5=5, 11-12-2023, INPROGRESS, NORMAL, [READING]]
System.out.println(collect.get(Boolean.FALSE));
```

But your solution is more human readable.